File: UploadFilesTxt.js
const fs = require('fs');
const path = require('path');
const axios = require('axios');

// Configuración de Azure Search
const vectorStoreAddress = "AzureEndpoint";
const vectorStorePassword = "AzureKey";
const indexName = "indice2";

// Función para cargar todos los archivos de texto de una carpeta
function loadDocumentsFromFolder(folderPath) {
  const documents = [];
  const files = fs.readdirSync(folderPath);
  files.forEach(file => {
    if (file.endsWith('.txt')) {
      const filePath = path.join(folderPath, file);
      const content = fs.readFileSync(filePath, 'utf8');
      documents.push(content);
    }
  });
  return documents;
}

// Función para dividir el texto en partes más pequeñas
function splitDocuments(documents, chunkSize = 1000) {
  const chunks = [];
  documents.forEach(doc => {
    for (let i = 0; i < doc.length; i += chunkSize) {
      chunks.push(doc.substring(i, i + chunkSize));
    }
  });
  return chunks;
}

// Función para subir documentos al vector store de Azure Search
async function addDocumentsToVectorStore(docs) {
  const documents = docs.map((doc, index) => ({
    '@search.action': 'upload',
    id: `doc-${index}`,
    text: doc
    // Aquí puedes agregar más campos si es necesario
  }));

  await axios.post(
    `${vectorStoreAddress}/indexes/${indexName}/docs/index?api-version=2021-04-30-Preview`,
    {
      value: documents
    },
    {
      headers: {
        'Content-Type': 'application/json',
        'api-key': vectorStorePassword
      }
    }
  );
}

(async () => {
  try {
    // Ruta de la carpeta que contiene los archivos de texto
    const folderPath = "PDFsResult";

    // Cargar todos los documentos de texto de la carpeta
    const documents = loadDocumentsFromFolder(folderPath);
    console.log(`Se cargaron correctamente ${documents.length} documentos.`);

    // Dividir los documentos en partes más pequeñas
    const docs = splitDocuments(documents);

    // Agregar los documentos divididos al vector store
    await addDocumentsToVectorStore(docs);
    console.log("Los documentos se subieron correctamente al vector store.");
  } catch (e) {
    console.error(`Ocurrió un error durante la carga o subida de documentos: ${e.message}`);
  }
})();

File: UploadFilesTxt.py
import os
from langchain_community.vectorstores.azuresearch import AzureSearch
from langchain_openai import OpenAIEmbeddings
from langchain_community.document_loaders import TextLoader
from langchain_text_splitters import CharacterTextSplitter

# Configuración de la API de OpenAI y Azure Search
openai_api_key = "key"
openai_api_version = "2023-05-15"
model = "text-embedding-ada-002"
vector_store_address = "AzureEndpoint"
vector_store_password = "AzureKey"

# Configuración de Azure Search
index_name = "indice2"

# Crear instancias de OpenAIEmbeddings y AzureSearch
embeddings = OpenAIEmbeddings(
    openai_api_key=openai_api_key,
    openai_api_version=openai_api_version,
    model=model
)

# Crear instancia de AzureSearch
vector_store = AzureSearch(
    azure_search_endpoint=vector_store_address,
    azure_search_key=vector_store_password,
    index_name=index_name,
    embedding_function=embeddings.embed_query,
)

# Función para cargar todos los archivos de texto de una carpeta
def load_documents_from_folder(folder_path):
    documents = []
    for filename in os.listdir(folder_path):
        if filename.endswith(".txt"):
            file_path = os.path.join(folder_path, filename)
            loader = TextLoader(file_path, encoding="utf-8")
            documents.extend(loader.load())
    return documents

# Ruta de la carpeta que contiene los archivos de texto
folder_path = "PDFsResult"

try:
    # Cargar todos los documentos de texto de la carpeta
    documents = load_documents_from_folder(folder_path)
    print(f"Se cargaron correctamente {len(documents)} documentos.")

    # Dividir los documentos en partes más pequeñas
    text_splitter = CharacterTextSplitter(chunk_size=1000, chunk_overlap=0)
    docs = text_splitter.split_documents(documents)

    # Agregar los documentos divididos al vector store
    vector_store.add_documents(documents=docs)
    print("Los documentos se subieron correctamente al vector store.")
except Exception as e:
    print(f"Ocurrió un error durante la carga o subida de documentos: {e}")
